js/tidex2.js

```diff
@@ -0,0 +1,960 @@
+                'active': undefined,
+                'contract': false,
+                'linear': undefined,
+                'inverse': undefined,
+                'taker': undefined,
+                'contractSize': undefined,
+                'expiry': undefined,
+                'expiryDatetime': undefined,
+                'strike': undefined,
+                'optionType': undefined,
+                'precision': precision,
+                'limits': {
+                    'leverage': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'amount': {
+                        'min': this.safeNumber (market, 'minAmount'),
+                        'max': undefined,
+                    },
+                    'price': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'cost': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                },
+                'info': market,
+            });
+        }
+        return result;
+    }
+
+    parseTicker (ticker, market = undefined) {
+        //
+        // fetchTicker
+        //     {
+        //         "name":"YFI_USDT",
+        //         "bid":"21607.842223",
+        //         "ask":"21635.50514299",
+        //         "open":"21530.31",
+        //         "high":"22619.38",
+        //         "low":"21193.59",
+        //         "last":"21644.23",
+        //         "volume":"188.4555124",
+        //         "deal":"4075612.142786252",
+        //         "change":"1"
+        //     }
+        //
+        // fetchTickers
+        //     {
+        //         "at":1646289676,
+        //         "ticker":{
+        //             "name":"yfi_usdt",
+        //             "bid":"21937.47155034",
+        //             "ask":"21966.13380147",
+        //             "open":"21449.08",
+        //             "high":"22619.38",
+        //             "low":"21193.59",
+        //             "last":"21968.03",
+        //             "vol":"188.4452385",
+        //             "deal":"4075398.065557751",
+        //             "change":"2"
+        //         }
+        //     }
+        //
+        const timestamp = this.safeTimestamp (ticker, 'at');
+        if (timestamp !== undefined) {
+            ticker = this.safeValue (ticker, 'ticker');
+        }
+        const marketId = this.safeStringUpper (ticker, 'name');
+        const symbol = this.safeSymbol (marketId, market);
+        const last = this.safeString (ticker, 'last');
+        return this.safeTicker ({
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': this.safeString (ticker, 'high'),
+            'low': this.safeString (ticker, 'low'),
+            'bid': this.safeString (ticker, 'buy'),
```

Comment:
it seems like we should change the lines 349 and 351 from
```JavaScript
'bid': this.safeString (ticker, 'buy'), 
...
'ask': this.safeString (ticker, 'sell'),
```
to >>>>
```JavaScript
'bid': this.safeString (ticker, 'bid'),
...
'ask': this.safeString (ticker, 'ask')
```

because `this.safeString (object, key)` extracts a value from the `object` which corresponds to the `key` and parses it into a string